src/ini/config.h

```
@@ -0,0 +1,22 @@
+#include "ini.h"
+
+
+typedef struct { /*Can be changed in the future*/
+        const char *orchestrator_ip;
+        const char *ochestrator_port;
+} configurationNode;
+
+typedef struct {
+        const char *port;
+        const char *expected_nodes;
+} configurationOrch;
```

I've talked to Artiom yesterday and personally I don't like creating predefined struct mapped to concrete ini file content. Config files are changed quite often, I think using a map, where each config file option will be a key, would be a better way to model config file content.
Artiom is going to provide more detailed description of the solution to discuss ...

---

I'd actually prefer predefined structs which gets its values mapped from an external datasource (an ini file in our case). It keeps the applications "internals" a bit better decoupled from the "outside" - its a contract. It also facilitates a "fail early" approach in the mapping (with a specific location in the code), e.g. if the passed config is invalid due to a missing port. In addition, the struct may be defining more complex fields than are representable in the ini file. For example:
```c
// instead of this:
typedef struct {
        const char *orchestrator_ip;
        const char *ochestrator_port;
} configurationNode;

// define it like this:
typedef struct {
        const struct sockaddr_in host;
} configurationNode;
```
The ini file definition stays the same (two key-value-pair for port and ip). The latter contains the same information, but represented differently and more suitable for later use in the application (although in this example its quite negligible).

---

I made the changes that Martin and I talked about. Now the parsing function will parse the data into a hashmap containing a struct you can view in the code. It works well. Feel free to view and commit the code.